src/main/java/com/titanic/airbnbclone/service/AccommodationService.java

```diff
@@ -86,13 +87,47 @@ public DeleteReservationResponseDto delete(Long accommodationId, Long reservatio
 
             return DeleteReservationResponseDto.builder()
                     .status(StatusEnum.SUCCESS.getStatusCode())
-                    .message(ReservationEnum.RESERVATION_CANCEL_SUCCESS.getMessage())
+                    .message(ReservationMessage.RESERVATION_CANCEL_SUCCESS.getMessage())
                     .build();
 
         } catch (Exception e) {
             return DeleteReservationResponseDto.builder()
                     .status(StatusEnum.ACCEPTED.getStatusCode())
-                    .message(ReservationEnum.RESERVATION_CANCEL_FAIL.getMessage())
+                    .message(ReservationMessage.RESERVATION_CANCEL_FAIL.getMessage())
+                    .build();
+        }
+    }
+
+    public ReservationInfoResponseDtoList getReservationInfo(HttpServletRequest request) {
+
+        try {
+            String userEmail = (String) request.getAttribute(OauthEnum.USER_EMAIL.getValue());
+            Account foundAccount = accountRepository.findByEmail(userEmail);
+            List<ReservationInfoResponseDto> allData = new ArrayList<>();
+
+            // 현재 로그인된 사용자의 예약 내역을 확인해 DTO로 만드는 과정
+           for (Reservation reservation : foundAccount.getReservations()) {
+                Long accommodationId = reservation.getAccommodation().getId();
+                Accommodation foundAccommodation = accommodationRepository.findOneWithPictures(accommodationId)
+                        .orElseThrow(() -> new NoSuchEntityException(accommodationId));
```

Comment:
여기서 객체그래프를 두번정도 갑니다. 이것보단 일급컬렉션을 사용해서 하나의 메서드로 가져오면 좋지 않을까요?
```suggestion
           for (Reservation reservation : foundAccount.getReservations()) {
                Long accommodationId = reservation.getAccomdationId;
                Accommodation foundAccommodation = accommodationRepository.findOneWithPictures(accommodationId)
                        .orElseThrow(() -> new NoSuchEntityException(accommodationId));
```

Reply:
해당 메서드가 일을 너무 많이 합니다. Account 가져오랴 예약내역도 가져오고 Dto도 만듭니다. 메서드를 각각 만들어주세요

Reply:
[일급컬렉션](https://woowacourse.github.io/javable/2020-05-08/First-Class-Collection)

Reply:
1. 일급컬렉션 피드백은 좋네요! 한번도 고려해보지 않은 사항이라 신선하네요. 코드의 중복을 줄이고 역할을 해당 컬렉션에게 넘긴다는 장점도 있는 것 같습니다. 

2. 메서드가 역할이 너무 많은지 한번 판단해보고 나눠야할지 판단해볼게요!